hypertrace-ingester/src/test/java/org/hypertrace/migration/MigrationTestRpc.java

```diff
@@ -0,0 +1,335 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class MigrationTestRpc {
+
+  private final Random random = new Random();
+  private JaegerSpanNormalizer normalizer;
+
+  @BeforeEach
+  public void setup()
+      throws SecurityException, NoSuchFieldException, IllegalArgumentException,
+          IllegalAccessException {
+    // Since JaegerToRawSpanConverter is a singleton, we need to reset it for unit tests to
+    // recreate the instance.
+    Field instance = JaegerSpanNormalizer.class.getDeclaredField("INSTANCE");
+    instance.setAccessible(true);
+    instance.set(null, null);
+
+    // Create a JaegerSpanNormaliser
+    String tenantId = "tenant-" + this.random.nextLong();
+    Map<String, Object> configs = Map.of("processor", Map.of("defaultTenantId", tenantId));
+    this.normalizer = JaegerSpanNormalizer.get(ConfigFactory.parseMap(configs));
+  }
+
+  @Test
+  public void testGrpcFields() throws Exception {
+
+    Map<String, String> tagsMap =
+        new HashMap<>() {
+          {
+            put(RawSpanConstants.getValue(GRPC_ERROR_MESSAGE), "Some error message");
+            put(RawSpanConstants.getValue(CENSUS_RESPONSE_STATUS_CODE), "12");
+            put(RawSpanConstants.getValue(GRPC_STATUS_CODE), "13");
+            put(RawSpanConstants.getValue(CENSUS_RESPONSE_CENSUS_STATUS_CODE), "14");
+            put(
+                RawSpanConstants.getValue(CENSUS_RESPONSE_STATUS_MESSAGE),
+                "CENSUS_RESPONSE_STATUS_MESSAGE");
+            put(RawSpanConstants.getValue(ENVOY_GRPC_STATUS_MESSAGE), "ENVOY_GRPC_STATUS_MESSAGE");
+            put(RawSpanConstants.getValue(GRPC_REQUEST_BODY), "some grpc request body");
+            put(RawSpanConstants.getValue(GRPC_RESPONSE_BODY), "some grpc response body");
+          }
+        };
+
+    Span span = createSpanFromTags(tagsMap);
+    RawSpan rawSpan = normalizer.convert("tenant-key", span);
+
+    assertAll(
+        () ->
+            assertEquals(
+                rawSpan.getEvent().getGrpc().getResponse().getErrorMessage(),
+                RpcSemanticConventionUtils.getGrpcErrorMsg(rawSpan.getEvent())),
+        () ->
+            assertEquals(
+                rawSpan.getEvent().getGrpc().getResponse().getStatusCode(),
+                RpcSemanticConventionUtils.getGrpcStatusCode(rawSpan.getEvent())),
+        () ->
+            assertEquals(
+                rawSpan.getEvent().getGrpc().getResponse().getStatusMessage(),
+                RpcSemanticConventionUtils.getGrpcStatusMsg(rawSpan.getEvent())),
+        () ->
+            assertEquals(
+                rawSpan.getEvent().getGrpc().getResponse().getSize(),
+                RpcSemanticConventionUtils.getGrpcResponseSize(rawSpan.getEvent()).get()),
```

Comment:
The current def of RpcSemanticConventionUtils.getGrpcResponseSize.getGrpcResponseSize which is wrapped around `GRPC_RESPONSE_BODY`. This is incorrect, it hides `RPC_RESPONSE_BODY` if it present as alternative.
```
if (attributeValueMap.get(RawSpanConstants.getValue(GRPC_RESPONSE_BODY)) != null) {

      if (attributeValueMap.get(RawSpanConstants.getValue(ENVOY_RESPONSE_SIZE)) != null) {
        return Optional.of(
            Integer.parseInt(
                attributeValueMap.get(RawSpanConstants.getValue(ENVOY_RESPONSE_SIZE)).getValue()));
      } else if (attributeValueMap.get(RawSpanConstants.getValue(GRPC_RESPONSE_BODY)) != null) {
        String requestBody =
            attributeValueMap.get(RawSpanConstants.getValue(GRPC_RESPONSE_BODY)).getValue();
        return Optional.of(requestBody.length());
      } else if (attributeValueMap.get(RPC_RESPONSE_BODY.getValue()) != null) {
        String requestBody = attributeValueMap.get(RPC_RESPONSE_BODY.getValue()).getValue();
        return Optional.of(requestBody.length());
      }
    }
```

The code should have been,
```
public static Optional<Integer> getGrpcResponseSize(Event event) {
    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
      return Optional.empty();
    }

    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
      if (attributeValueMap.get(RawSpanConstants.getValue(ENVOY_RESPONSE_SIZE)) != null) {
        return Optional.of(
            Integer.parseInt(
                attributeValueMap.get(RawSpanConstants.getValue(ENVOY_RESPONSE_SIZE)).getValue()));
      } else if (attributeValueMap.get(RawSpanConstants.getValue(GRPC_RESPONSE_BODY)) != null) {
        String requestBody =
            attributeValueMap.get(RawSpanConstants.getValue(GRPC_RESPONSE_BODY)).getValue();
        return Optional.of(requestBody.length());
      } else if (attributeValueMap.get(RPC_RESPONSE_BODY.getValue()) != null) {
        String requestBody = attributeValueMap.get(RPC_RESPONSE_BODY.getValue()).getValue();
        return Optional.of(requestBody.length());
      }
    return Optional.empty();
  }
 ```

Reply:
This was also present in both `fieldGeneratorMap` and in `rpcFieldGeneratorMap` 🙈